fix tests that could have racy script module instantiation

zhaojuanmao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#34792 fix tests that could have racy script module instantiation**

it is thread safe to initiate script module in multiple threads.

for both test_remote_script_module and test_torchscript_functions_not_supported, it is possible that client thread is initiating MyScriptModule while server thread is initiating it as well in the same rank process.

removing MyScriptModule instatiation in client thread, it is not needed actually.

Differential Revision: [D20463234](https://our.internmc.facebook.com/intern/diff/D20463234/)

zhaojuanmao committed:
fix tests that could have racy script module instantiation

it is thread safe to initiate script module in multiple threads.

for both test_remote_script_module and test_torchscript_functions_not_supported, it is possible that client thread is initiating MyScriptModule while server thread is initiating it as well in the same rank process.

removing MyScriptModule instatiation in client thread, it is not needed actually.

Differential Revision: [D20463234](https://our.internmc.facebook.com/intern/diff/D20463234/)

[ghstack-poisoned]

zhaojuanmao committed:
Update on "fix tests that could have racy script module instantiation"


it is thread safe to initiate script module in multiple threads.

for both test_remote_script_module and test_torchscript_functions_not_supported, it is possible that client thread is initiating MyScriptModule while server thread is initiating it as well in the same rank process.

removing MyScriptModule instatiation in client thread, it is not needed actually.

Differential Revision: [D20463234](https://our.internmc.facebook.com/intern/diff/D20463234/)

[ghstack-poisoned]

pritamdamania87 commented on torch/testing/_internal/distributed/rpc/jit/rpc_test.py:
```diff
@@ -626,7 +635,7 @@ def test_remote_script_module(self):
         api._ignore_rref_leak = True
 
-        local_ret = MyScriptModule(self.rank).forward() + torch.ones(self.rank)
+        local_ret = torch.ones(self.rank) + torch.ones(self.rank)
```
nit: Can we do the same here?

zhaojuanmao committed:
Update on "fix tests that could have racy script module instantiation"


it is thread safe to initiate script module in multiple threads.

for both test_remote_script_module and test_torchscript_functions_not_supported, it is possible that client thread is initiating MyScriptModule while server thread is initiating it as well in the same rank process.

removing MyScriptModule instatiation in client thread, it is not needed actually.

Differential Revision: [D20463234](https://our.internmc.facebook.com/intern/diff/D20463234/)

[ghstack-poisoned]

zhaojuanmao committed:
Update on "fix tests that could have racy script module instantiation"


it is thread safe to initiate script module in multiple threads.

for both test_remote_script_module and test_torchscript_functions_not_supported, it is possible that client thread is initiating MyScriptModule while server thread is initiating it as well in the same rank process.

removing MyScriptModule instatiation in client thread, it is not needed actually.

Differential Revision: [D20463234](https://our.internmc.facebook.com/intern/diff/D20463234/)

[ghstack-poisoned]
